src/wallet/rpcwallet.cpp

```diff
@@ -77,6 +77,20 @@ static bool ParseIncludeWatchonly(const UniValue& include_watchonly, const CWall
 }
 
 
+static CWallet* GetReadyWallet(std::shared_ptr<CWallet> const wallet) {
```

Comment:
d37756f380b27d259a7818078ddf02caa0a4578a

NAK the dereferencing because the wallet can be destroyed after the call, like
```cpp
CWallet* const pwallet = GetReadyWallet(GetWalletForJSONRPCRequest(request));
```
Also it's not clear the function does the dereference.

Reply:
Maybe the following signature might be better?

```
 static CWallet& GetReadyWallet(CWallet& wallet)
{